product_supplierinfo_for_customer_sale_quick/models/product_product.py

```diff
@@ -0,0 +1,28 @@
+from odoo import api, fields, models
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    product_customer_code = fields.Char(
+        related="variant_customer_ids.product_code",
+        string="Product Customer Code",
+        readonly=True,
+    )
+
+    product_customer_name = fields.Char(
+        related="variant_customer_ids.product_name",
+        string="Product Customer Name",
+        readonly=True,
+    )
```

Comment:
**issue**: unfortunately these won't do, as the `related` field will always only take the first record it finds from `variant_customer_ids`, which may not be the one we want. 

In module product_supplierinfo_for_customer there's a function `_select_customerinfo()` https://github.com/OCA/product-attribute/blame/14.0/product_supplierinfo_for_customer/models/product_product.py#L122 that returns the correct `product.customerinfo` based on partner, quantity etc. (though that function could use a couple of fixes as well, but that's for another PR)